code/obj/machinery/flasher.dm

```diff
@@ -227,14 +218,56 @@
 
 	return
 
+/obj/machinery/flasher/portable //Portable version of the flasher. Only flashes when anchored
+	name = "portable flasher"
+	desc = "A portable flashing device. Wrench to activate and deactivate. Cannot detect slow movements."
+	icon_state = "pflash1-c"
+	strength = 8
+	anchored = 0
+	base_state = "pflash"
+	density = 1
+	event_handler_flags = USE_PROXIMITY | USE_FLUID_ENTER
+	var/cooldown_flash = 15 SECONDS
+	var/cooldown_scan = 1.5 SECONDS
+	var/cooldown_end = 0
+
+/obj/machinery/flasher/portable/power_change()
+	..()
+	if (powered())
+		if (!src.anchored)
+			icon_state = "[base_state]1-c"
+		else
+			if (src.last_flash && world.time < src.cooldown_end)
+				icon_state = "[base_state]1-c"
+			light.enable()
+
 /obj/machinery/flasher/portable/HasProximity(atom/movable/AM as mob|obj)
-	if ((src.disable) || (src.last_flash && world.time < src.last_flash + 150))
+	if (!src.anchored || src.disable || (src.last_flash && world.time < src.cooldown_end))
+		return
+
+	if (!powered())
```

Comment:
I would expect there is less overhead keeping the powered check behind the simple variable checks considering HasProximity can be called multiple times a second.